Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -1056,7 +1063,82 @@ void PcapLiveDevice::setDefaultGateway()
 			PCPP_LOG_ERROR("Error retrieving default gateway address: " << e.what());
 		}
 	}
-#elif defined(__APPLE__) || defined(__FreeBSD__)
+#elif defined(__APPLE__)
+
+	//route message struct for communication in APPLE device
+	struct BSDRoutingMessage
+	{
+		struct rt_msghdr header;
+		char   messageSpace[512];
+	};
+
+	struct BSDRoutingMessage routingMessage;
+	// It creates a raw socket that can be used for routing-related operations
+	int sockfd = socket(PF_ROUTE, SOCK_RAW, 0);
+	if (sockfd < 0)
+	{
+		PCPP_LOG_ERROR("Error retrieving default gateway address: couldn't get open routing socket");
+		return ;
+	}
+	memset(reinterpret_cast<char*>(&routingMessage), 0, sizeof(routingMessage));
+	routingMessage.header.rtm_msglen = sizeof(struct rt_msghdr);
+	routingMessage.header.rtm_version = RTM_VERSION;
+	routingMessage.header.rtm_type = RTM_GET;
+	routingMessage.header.rtm_addrs = RTA_DST | RTA_NETMASK ;
+	routingMessage.header.rtm_flags = RTF_UP | RTF_GATEWAY | RTF_STATIC;
+	routingMessage.header.rtm_msglen += 2 * sizeof(sockaddr_in);
+
+	if (write(sockfd, reinterpret_cast<char*>(&routingMessage), routingMessage.header.rtm_msglen) < 0)
+	{
+		PCPP_LOG_ERROR("Error retrieving default gateway address: couldn't write into the routing socket");
+		return;
+	}
+
+	// Read the response from the route socket
+	if (read(sockfd, reinterpret_cast<char*>(&routingMessage), sizeof(routingMessage)) < 0)
+	{
+		PCPP_LOG_ERROR("Error retrieving default gateway address: couldn't read from the routing socket");
+		return;
+	}
+
+	struct in_addr  *gateAddr = nullptr;
+	struct sockaddr *sa = nullptr;
+	char* spacePtr = (reinterpret_cast<char*>(&routingMessage.header + 1));
+	auto rtmAddrs = routingMessage.header.rtm_addrs;
+	int index = 1;
+	auto roundUpClosestMultiple = [](int multiple, int num) {
+		return ((num+multiple - 1) / multiple) * multiple;
+	};
+	while (rtmAddrs)
```

Comment:
Not really an issue, but I feel like moving the lambda `roundUpClosestMultiple` above the variable declarations might be beneficial for readability. IMO, it currently breaks the flow between the variables used in the `while` loop and the loop itself. No strong opinion on if the suggestion is implemented or not, but I thought I might mention it.

Reply:
but the lambda is also used in the while loop?

Reply:
My thought was that the lambda, just like some variable declarations, is only used in the next while loop, so I defined it right above the loop. I think we can keep it as it is.

Reply:
Sure. As I said, no strong opinion.